independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java

```diff
@@ -0,0 +1,88 @@
+package io.quarkus.arc.test.buildextension.beans;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+import jakarta.enterprise.util.TypeLiteral;
+import jakarta.inject.Singleton;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.ParameterizedType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.All;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.BeanCreator;
+import io.quarkus.arc.SyntheticCreationalContext;
+import io.quarkus.arc.processor.BeanRegistrar;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class SyntheticInjectionPointListAllTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(SomeBean.class)
+            .beanRegistrars(new TestRegistrar()).build();
+
+    @Test
+    public void testListAllInjection() {
+        SyntheticBean synthBean = Arc.container().instance(SyntheticBean.class).get();
+        assertNotNull(synthBean);
+        List<SomeBean> list = synthBean.getList();
+        assertNotNull(list);
+        assertTrue(list.size() == 1);
```

Comment:
`assertEquals(1, list.size())` is usually a bit better - at least you can see the actual size in the log ;-)